pkg/distributors/test/MerkleRedeem.test.ts

```diff
@@ -0,0 +1,273 @@
+    await tokens.approve({ to: vault.address, from: [lp1] });
+
+    const assets = tokens.addresses;
+    const assetManagers = [mockAssetManager.address, mockAssetManager.address];
+    await pool.registerTokens(assets, assetManagers);
+
+    await vault.connect(lp1).joinPool(poolId, lp1.address, lp2.address, {
+      assets,
+      maxAmountsIn: assets.map(() => MAX_UINT256),
+      fromInternalBalance: false,
+      userData: encodeJoin(
+        assets.map(() => tokenInitialBalance),
+        assets.map(() => 0)
+      ),
+    });
+  });
+
+  it('stores an allocation', async () => {
+    const claimBalance = bn('9876');
+
+    const elements = [encodeElement(lp1.address, claimBalance)];
+    const merkleTree = new MerkleTree(elements);
+    const root = merkleTree.getHexRoot();
+
+    await merkleRedeem.connect(admin).seedAllocations(bn(1), root, claimBalance);
+
+    const proof = merkleTree.getHexProof(elements[0]);
+
+    const result = await merkleRedeem.verifyClaim(lp1.address, 1, claimBalance, proof);
+    expect(result).to.equal(true);
+  });
+
+  it("doesn't allow an allocation to be overwritten", async () => {
+    const claimBalance = bn('9876');
+
+    const elements = [encodeElement(lp1.address, claimBalance)];
+    const merkleTree = new MerkleTree(elements);
+    const root = merkleTree.getHexRoot();
+
+    await merkleRedeem.seedAllocations(1, root, claimBalance);
+
+    // construct tree to attempt to override the allocation
+    const elements2 = [encodeElement(lp1.address, claimBalance), encodeElement(lp2.address, claimBalance)];
+    const merkleTree2 = new MerkleTree(elements2);
+    const root2 = merkleTree2.getHexRoot();
+
+    const errorMsg = 'cannot rewrite merkle root';
+    expect(merkleRedeem.seedAllocations(1, root2, claimBalance.mul(2))).to.be.revertedWith(errorMsg);
+  });
+
+  it('stores multiple allocations', async () => {
+    const claimBalance0 = bn('1000');
+    const claimBalance1 = bn('2000');
+
+    const elements = [encodeElement(lp1.address, claimBalance0), encodeElement(lp2.address, claimBalance1)];
+    const merkleTree = new MerkleTree(elements);
+    const root = merkleTree.getHexRoot();
+
+    await merkleRedeem.seedAllocations(1, root, bn('3000'));
+
+    const proof0 = merkleTree.getHexProof(elements[0]);
+    let result = await merkleRedeem.verifyClaim(lp1.address, 1, claimBalance0, proof0);
+    expect(result).to.equal(true); //"account 0 should have an allocation";
+
+    const proof1 = merkleTree.getHexProof(elements[1]);
+    result = await merkleRedeem.verifyClaim(lp2.address, 1, claimBalance1, proof1);
+    expect(result).to.equal(true); // "account 1 should have an allocation";
+  });
+
+  describe('When a user has an allocation to claim', () => {
+    const claimBalance = bn('1000');
+    let elements: string[];
+    let merkleTree: MerkleTree;
+
+    beforeEach(async () => {
+      elements = [encodeElement(lp2.address, claimBalance)];
```

Comment:
```suggestion
    const week = 1;
    let elements: string[];
    let merkleTree: MerkleTree;
    let lp: SignerWithAddress;

    beforeEach(async () => {
      lp = lp2.address;
      elements = [encodeElement(lp2.address, claimBalance)];
```

I think this will make the tests easier to read.